src/main/scala/is/hail/expr/Type.scala

```diff
@@ -808,6 +823,15 @@ abstract class TContainer extends Type {
       }
     }
   }
+
+  def genElement: Gen[Annotation] = if (elementsRequired) elementType.genNonmissingValue else elementType.genValue
+
+  def typeCheckElement(a: Any): Boolean = {
+    if (elementsRequired)
```

Comment:
Put the if condition inside the assert: `assert(!elementsRequired || a != null, ...)`.